lib/Dialect/TTIR/IR/TTIROps.cpp

```diff
@@ -126,6 +126,36 @@ ::mlir::LogicalResult mlir::tt::ttir::TransposeOp::verify() {
   return success();
 }
 
+::mlir::LogicalResult mlir::tt::ttir::ReshapeOp::verify() {
+  ::mlir::RankedTensorType outputType = getOutput().getType();
+  auto shape = getShape();
+  int64_t shape_size = static_cast<int64_t>(shape.size());
+
+  if (shape_size == 0) {
+    return emitOpError("Shape attribute must be non-empty");
+  }
+
+  if (shape_size > 5) {
+    return emitOpError("Shape attribute must have at most 5 elements");
+  }
+
+  if (shape_size != outputType.getRank()) {
```

Comment:
Is this accurate statement? E.g. squeeze is a type of reshape that will reduce tensor rank.

Reply:
@mtopalovicTT you could call `.getNumElements()` on input and output tensors, and assert that they match.

Reply:
Thanks for pointing that out. I added volume check and removed rank check